### PR TITLE
Ti 3250 improve booking indicator on floorplan

### PR DESCRIPTION
**Pull request by accioweit**
https://thing-it.atlassian.net/browse/TI-3250

### CODE REVIEW

File: projects/thing-it-bim-lib/src/lib/bookings-renderer/bookings-renderer.component.ts [hunk 1]
```diff
@@ -120,11 +26,19 @@ export class BookingsRendererComponent implements OnInit, OnChanges, OnDestroy {
 
   @Input()
   set data(_data) {
+    _data.forEach(el => {
```
Comment (aleksey-trotsiuk) [inline projects/thing-it-bim-lib/src/lib/bookings-renderer/bookings-renderer.component.ts]: 1. you should check `_data` property on `null`, and `undefined`. same should be applied for `el.bookings`
2. you should prevent filtering if the same `_data` comes.
3.  please, check whether you can get rid of using moment `isBefore`, `isAfter` and `isSameOrAfter`. It ay affects performance on iOS

File: projects/thing-it-bim-lib/src/lib/bookings-renderer/bookings-renderer.component.ts [hunk 2]
```diff
@@ -240,9 +154,9 @@ export class BookingsRendererComponent implements OnInit, OnChanges, OnDestroy {
 
     const unitsInInterval = 720;
     const minuteAngle = 360 / unitsInInterval;
-    const startHour = this._options.startHour ? this._options.startHour : 7;
+    const startHour = this._options.startHour ? this._options.startHour : moment().hours();
```
Comment (aleksey-trotsiuk) [inline projects/thing-it-bim-lib/src/lib/bookings-renderer/bookings-renderer.component.ts]: `moment().hours()`  can be moved to const

File: projects/thing-it-bim-lib/src/lib/bookings-renderer/bookings-renderer.component.ts [hunk 3]
```diff
@@ -120,11 +26,28 @@ export class BookingsRendererComponent implements OnInit, OnChanges, OnDestroy {
 
   @Input()
   set data(_data) {
-    this._data = _data;
-    this.render();
+    if (_data){
```
Comment (Malik19051992) [inline projects/thing-it-bim-lib/src/lib/bookings-renderer/bookings-renderer.component.ts]: add an empty line before if, after if and before return. Also add types

File: projects/thing-it-bim-lib/src/lib/bookings-renderer/bookings-renderer.component.ts [hunk 4]
```diff
@@ -120,11 +26,28 @@ export class BookingsRendererComponent implements OnInit, OnChanges, OnDestroy {
...
+      const minDate = moment().toDate();
+      const maxDate = moment().add(12, 'hours').toDate();
+      _data.forEach(el => {
+        if(el.bookings){
+        return el.bookings = el.bookings.filter(booking => {
```
Comment (Malik19051992) [inline projects/thing-it-bim-lib/src/lib/bookings-renderer/bookings-renderer.component.ts]: use lodash

File: projects/thing-it-bim-lib/src/lib/bookings-renderer/bookings-renderer.component.ts [hunk 5]
```diff
@@ -120,11 +26,28 @@ export class BookingsRendererComponent implements OnInit, OnChanges, OnDestroy {
...
+            let bookingFrom = moment(booking.from).toDate();
+            let bookingTo = moment(booking.to).toDate();
+            if (bookingFrom < minDate && bookingTo > minDate && maxDate >= bookingTo){
+              booking.from = moment();
+              bookingFrom = moment().toDate();              
+            }
+            return bookingFrom >= minDate && maxDate >= bookingTo;
```
Comment (Malik19051992) [inline projects/thing-it-bim-lib/src/lib/bookings-renderer/bookings-renderer.component.ts]: what about situation: 
minDate: 12:00
maxDate: 24:00
booking from: 23:00
booking to: 15:00 next day

I thing need to cut it

File: projects/thing-it-bim-lib/src/lib/bookings-renderer/bookings-renderer.component.ts [hunk 6]
```diff
@@ -120,11 +26,28 @@ export class BookingsRendererComponent implements OnInit, OnChanges, OnDestroy {
...
+          }); 
+        }
+      });
+      this._data = _data;
+      this.render();
+    }
   }
 
-  get data(): string {
+  get data() {
```
Comment (Malik19051992) [inline projects/thing-it-bim-lib/src/lib/bookings-renderer/bookings-renderer.component.ts]: why drop type?

File: projects/thing-it-bim-lib/src/lib/bookings-renderer/bookings-renderer.component.ts [hunk 7]
```diff
@@ -26,28 +27,38 @@ export class BookingsRendererComponent implements OnInit, OnChanges, OnDestroy {
...
-            let bookingFrom = moment(booking.from).toDate();
-            let bookingTo = moment(booking.to).toDate();
-            if (bookingFrom < minDate && bookingTo > minDate && maxDate >= bookingTo){
-              booking.from = moment();
-              bookingFrom = moment().toDate();              
-            }
-            return bookingFrom >= minDate && maxDate >= bookingTo; 
-          }); 
-        }
+          el.bookings = _.filter(el.bookings, function (booking: any) {
```
Comment (Malik19051992) [inline projects/thing-it-bim-lib/src/lib/bookings-renderer/bookings-renderer.component.ts]: do please refactoring. 
`_(el.bookings)
.filter((booking: any)=>...  )
.map((booking: any)=>... )`

**Pull request merged**